public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -204,11 +210,30 @@ export function GrafanaEvaluationBehaviorStep({
 
   const isNewGroup = useIsNewGroup(folderUid ?? '', group);
 
+  const handleFieldsDependecies: WatchObserver<RuleFormValues> = useCallback(
```

Comment:
Excercies: I do wonder if having one function handling dependencies could be easier to read/write than multiple watches and effects

Reply:
Personally, i believe it would be easier to understand/maintain in a single place, also i can avoid duplication of useEffects observing the same fields..

Reply:
Yeah, this is the idea to remove `useEffects`